packages/compat/src/fixup-rules.js

```diff
@@ -86,118 +56,145 @@ export function fixupRule(ruleDefinition) {
 		return fixedUpRuleReplacements.get(ruleDefinition);
 	}
 
-	const isLegacyRule = typeof ruleDefinition === "function";
-
 	// check to see if this rule definition has already been fixed up
-	if (!isLegacyRule && fixedUpRules.has(ruleDefinition)) {
+	if (fixedUpRules.has(ruleDefinition)) {
 		return ruleDefinition;
 	}
 
-	const originalCreate = isLegacyRule
-		? ruleDefinition
-		: ruleDefinition.create.bind(ruleDefinition);
+	const originalCreate = ruleDefinition.create.bind(ruleDefinition);
 
 	function ruleCreate(context) {
-		// if getScope is already there then no need to create old methods
-		if ("getScope" in context) {
+		// if getCwd is already there then no need to create old methods
+		if ("getCwd" in context) {
 			return originalCreate(context);
 		}
 
 		const sourceCode = context.sourceCode;
-		let currentNode = sourceCode.ast;
 
-		const newContext = Object.assign(Object.create(context), {
-			parserServices: sourceCode.parserServices,
-
-			/*
-			 * The following methods rely on the current node in the traversal,
-			 * so we need to add them manually.
-			 */
-			getScope() {
-				return sourceCode.getScope(currentNode);
+		const newSourceCode = Object.assign(Object.create(sourceCode), {
+			getTokenOrCommentBefore(node, skip) {
+				return sourceCode.getTokenBefore(node, {
+					includeComments: true,
+					skip,
+				});
 			},
-
-			getAncestors() {
-				return sourceCode.getAncestors(currentNode);
+			getTokenOrCommentAfter(node, skip) {
+				return sourceCode.getTokenAfter(node, {
+					includeComments: true,
+					skip,
+				});
 			},
-
-			markVariableAsUsed(variable) {
-				sourceCode.markVariableAsUsed(variable, currentNode);
+			isSpaceBetweenTokens(first, second) {
+				return sourceCode.isSpaceBetween(first, second);
+			},
+			getJSDocComment(node) {
+				function isCommentToken(token) {
+					return ["Block", "Line", "Shebang"].includes(token.type);
+				}
+
+				function looksLikeExport(astNode) {
+					return (
+						astNode.type === "ExportDefaultDeclaration" ||
+						astNode.type === "ExportNamedDeclaration" ||
+						astNode.type === "ExportAllDeclaration" ||
+						astNode.type === "ExportSpecifier"
+					);
+				}
+
+				function findJSDocComment(astNode) {
+					const tokenBefore = sourceCode.getTokenBefore(astNode, {
+						includeComments: true,
+					});
+
+					if (
+						tokenBefore &&
+						isCommentToken(tokenBefore) &&
```

Comment:
The `isCommentToken` check is unnecessary as the next condition checks for the specific `type === Block`.